packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -610,3 +612,24 @@ export async function collectFilesForIndex(
     // pick top 100k files below size limit
     return storage.slice(0, Math.min(100000, i))
 }
+
+/**
+ * Performs a case-insensitive, recursive search for a string in a directory.
+ * note:
+ * 'error' is set when command fails to spawn; 'stderr' is set when command itself fails.
+ * 'exitCode' will be 0 when searchStr is detected (each line where it's found will be printed to 'stdout').
+ * 'exitCode' will be non-zero and 'stdout' / 'stderr' / 'error' will all be empty/undefined when searchStr is not detected.
+ * @param searchStr the string to search for
+ * @param dirPath the path of the directory to search in
+ * @returns the result of the search
+ */
+export async function findStringInDirectory(searchStr: string, dirPath: string) {
```

Comment:
might want some direct test coverage of this

Reply:
Yes see very bottom of PR, in `workspaceUtils.test.ts`